File: backend/app/schemas/analytics.py
import uuid
from typing import Optional
from pydantic import BaseModel


class HostelHealthResponse(BaseModel):
    """Flutter-compatible hostel health model. Field names match dormconnect_core HostelHealth."""
    hostel_id: str
    hostel_name: str
    health_score: float
    total_students: int
    current_out: int
    complaints_pending: int
    maintenance_pending: int
    pending_leaves: Optional[int] = None
    pending_enrollments: Optional[int] = None


class HostelStats(BaseModel):
    hostel_id: uuid.UUID
    hostel_name: str
    total_students: int
    active_students: int
    currently_out: int
    on_leave_today: int
    pending_complaints: int
    resolved_complaints: int
    total_complaints: int
    pending_maintenance: int
    health_score: float


class InstitutionOverview(BaseModel):
    institution_id: uuid.UUID
    institution_name: str
    total_students: int
    currently_out: int
    hostels: list[HostelStats]


class StaffPerformance(BaseModel):
    user_id: uuid.UUID
    name: str
    role: str
    complaints_resolved: int
    leaves_processed: int
    avg_response_hours: Optional[float] = None

File: mobile/packages/dormconnect_core/lib/src/models/analytics.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'analytics.freezed.dart';
part 'analytics.g.dart';

@freezed
class HostelHealth with _$HostelHealth {
  const factory HostelHealth({
    required String hostelId,
    required String hostelName,
    required double healthScore,
    required int totalStudents,
    required int currentOut,
    required int complaintsPending,
    required int maintenancePending,
    int? pendingLeaves,
    int? pendingEnrollments,
  }) = _HostelHealth;

  factory HostelHealth.fromJson(Map<String, dynamic> json) =>
      _$HostelHealthFromJson(json);
}

@freezed
class InstitutionOverview with _$InstitutionOverview {
  const factory InstitutionOverview({
    required String institutionName,
    required int totalStudents,
    required int totalOut,
    required int onLeaveToday,
    required int overdueReturns,
    required List<HostelHealth> hostels,
  }) = _InstitutionOverview;

  factory InstitutionOverview.fromJson(Map<String, dynamic> json) =>
      _$InstitutionOverviewFromJson(json);
}
